RFS/src/main/java/com/rfs/common/OpenSearchClient.java

```diff
@@ -310,6 +316,39 @@ public RestClient.Response refresh() {
         return client.get(targetPath);
     }
 
+    /*
+     * Retrieves documents from the specified index with the specified query.  Returns a list of the hits.
+     */
+    public List<ObjectNode> searchDocuments(String indexName, String queryBody) {
+        String targetPath = indexName + "/_search";
+        RestClient.Response response = client.postAsync(targetPath, queryBody.toString())
+            .flatMap(resp -> {
+                if (resp.code == HttpURLConnection.HTTP_OK) { 
+                    return Mono.just(resp);
+                } else {
+                    String errorMessage = ("Could not retrieve documents from index " + indexName + ". Response Code: " + resp.code
+                        + ", Response Message: " + resp.message + ", Response Body: " + resp.body);
+                    return Mono.error(new OperationFailed(errorMessage, resp));
+                }
+            })
+            .doOnError(e -> logger.error(e.getMessage()))
+            .retryWhen(Retry.backoff(3, Duration.ofSeconds(1)).maxBackoff(Duration.ofSeconds(10)))
+            .block();
```

Comment:
Seems like refactoring this would be useful to avoid unit testing code that is always the same, and would also reduce the magic number count - alongside limiting copypasta mistakes.

Reply:
Agreed, in theory.  However, it's unclear whether this is code we want to unit test vs. test at a higher level (such as with the dockerized "component" tests).  That said, if you have thoughts on how to refactor in order to make that unit testing easier, I'm all ears.  However, I personally don't think now is quite the time to do that.  I'd rather get the distributed behavior working first then circle back on stuff like this once we see how those higher level tests go.

Reply:
When I say testing I mean that you can know that things that should retry do it and do it in a consistent way.  By defining this retry logic on each use of client it would be easy for this to be missed, or altered in ways that are hard to detect.  

Verifying the library works isn't something I'd generally want to spend cycles testing - we'd just want its data plugged into our telemetry / monitoring story.